### ISSUE TITLE
netdata-updater.sh fails due to missing './' on 'netdata-installer.sh' line

### ISSUE THREAD

**Issue by jeremysherriff** [firehol/netdata#3940]
(I have trouble believing no one else has this problem :D )
Ubuntu 16.04 LTS, all patches and packages current.

netdata-updater.sh fails to run with a script error:
```
root@openflixr:/opt/netdata.git# ./netdata-updater.sh
Sat Jul  7 15:54:11 NZST 2018 : INFO:  Running on a terminal - (this script also supports running headless from crontab)

Sat Jul  7 15:54:11 NZST 2018 : INFO:  Updating netdata source from github...
Updating 93a5225..479f882
Fast-forward
 CONTRIBUTORS.md           |  4 ++++
 plugins.d/charts.d.plugin |  2 +-
 python.d/redis.chart.py   |  2 +-
 src/main.c                |  1 +
 src/rrdpush.c             | 28 ++++++++++++++++++++++++++++
 src/web_server.c          |  1 +
 src/web_server.h          |  1 +
 7 files changed, 37 insertions(+), 2 deletions(-)

Sat Jul  7 15:54:13 NZST 2018 : INFO:  Re-installing netdata...
./netdata-updater.sh: line 106: netdata-installer.sh: command not found
Sat Jul  7 15:54:13 NZST 2018 : ERROR:  FAILED TO UPDATE NETDATA : FAILED TO COMPILE/INSTALL NETDATA
```
If I edit line 106 of the file and add './' at the beginning of the line:
```
    ./netdata-installer.sh --install /opt  --dont-wait ${do_not_start} >&3 2>&3 || failed "FAILED TO COMPILE/INSTALL NETDATA"
```
it now "works fine" as the `git pull` has already completed and now reports nothing to do, but the installer line hasn't run to patch the code.
So I manually run `./netdata-installer.sh --install /opt --dont-wait` but I fear that I'm breaking stuff as I haven't exported the `NETDATA_CONFIGURE_OPTIONS` and `CFLAGS` variables etc.

After the next commit on github, the updater script will fail on the git pull due to local uncommitted changes (lol) meaning I then have to run with `-f` force option, overwriting the netdata-updater.sh file back to the one missing the './'.

Given that many people run netdata, and keep up to date successfully, I am really confused about this!
Is my system really the only one that can't run the updater script?

**Comment by ktsaou** [comment]
Hi,

the `netdata-updater.sh` script is generated by `netdata-installer.sh` every time it completes installing netdata.

The script generated, runs the installer the same way you run it in the first place. It uses these: https://github.com/firehol/netdata/blob/07f25d0ae94c7e4fd1fd8c4463c62d31f6a0be42/netdata-installer.sh#L78-L79

And when it generates the script it runs it like this: https://github.com/firehol/netdata/blob/07f25d0ae94c7e4fd1fd8c4463c62d31f6a0be42/netdata-installer.sh#L1251-L1252

So, I think you have this problem, because you run `netdata-installer.sh` without `./` when you first installed netdata. I admit this is quite unusual. Do you have `.` in your PATH?

**Comment by jeremysherriff** [comment]
Interesting. No, '.' is not in the path.  It wasn't me who did the install, but I suspect it may have been run using the context "bash netdata-installer.sh".

I assume I can amend the original stored variable (if I can find it)

**Comment by ktsaou** [comment]
hm... interesting... I didn't thought of that context...
So, this is a bug...

**Comment by jeremysherriff** [comment]
My machine is from a project called openflixr (www.openflixr.com) and the creator uses a weekly update script to force updates on a lot of installed packages, as the protection is supposed to be easy for first time Linux users who may not know how to do git pulls, set up Cron etc.  I have just found the entry for netdata:
```
## Netdata
echo ""
echo "Netdata:"
service netdata stop
killall netdata
cd /opt/netdata.git/
git reset --hard
sudo git pull
sudo bash netdata-installer.sh --dont-wait --install /opt
```
So arguably, the bug is in my side not in netdata (?)
The openflixr project seems to be abandoned, so it will be hard for me to get updates pushed for the weekly update script.  I will submit a PR anyway, but in the meantime it may be useful to address "other" ways of running the netdata-installer.sh script in case this issue affects others.

**Comment by ktsaou** [comment]
Thanks!
It is fixed in PR #3942

**Comment by ktsaou** [comment]
merged it.
On the next update, this problem is gone...
